### PR TITLE
Adds UI for shrink action

### PR DESCRIPTION
**Pull request by downsrob**
Signed-off-by: Clay Downs <downsrob@amazon.com>

### Description
Adds UI for the shrink action. This PR does not contain any tests, which should be added later. Additional validation should also be added, such as not allowing percentage to shrink shards to input out of the range of 0 to 1. This is currently validated by the backend but additional validation up front would improve the user experience.
### Zoomed out overview of the visual editor:
Screen Shot 2022-04-14 at 7 41 15 PM

### If more than one number of shards setting is added, this error appears:
Screen Shot 2022-04-14 at 7 41 49 PM

### When either of the JSON editors are poorly formatted it will show as red.
Screen Shot 2022-04-14 at 7 42 25 PM
If any of the above errors are present, it will not be possible to add the action.

### Template and alias updated to say -optional
Screen Shot 2022-04-15 at 3 46 12 PM

### Warning that appears for force unsafe option
Screen Shot 2022-04-15 at 4 29 26 PM

### Issues Resolved
https://github.com/opensearch-project/index-management/issues/40

### Check List
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/pages/VisualCreatePolicy/components/UIActions/ShrinkUIAction.tsx [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+    );
+  };
+
+  toAction = () => {
+    // Use the spread operator to copy the action
+    const shrink = { ...this.action };
+    if (shrink.aliases_json != null) {
+      shrink.shrink.aliases = JSON.parse(shrink.aliases_json as string);
+      console.log(shrink.aliases_json as string);
+      console.log((shrink.shrink.aliases as Object[])[0]);
```
Comment (thalurur) [inline public/pages/VisualCreatePolicy/components/UIActions/ShrinkUIAction.tsx:267]: remove?

File: public/pages/VisualCreatePolicy/components/UIActions/ShrinkUIAction.tsx [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+    }
+    if (shrink.target_index_name_template_json != null) {
+      shrink.shrink.target_index_name_template = JSON.parse(shrink.target_index_name_template_json as string);
+    }
+    delete shrink.aliases_json;
+    delete shrink.force_unsafe_input;
+    delete shrink.target_index_name_template_json;
+    console.log(shrink);
```
Comment (thalurur) [inline public/pages/VisualCreatePolicy/components/UIActions/ShrinkUIAction.tsx:275]: remove?

File: public/pages/VisualCreatePolicy/components/UIActions/ShrinkUIAction.tsx [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+          isInvalid={false}
+        />
+        <EuiFormRow fullWidth isInvalid={false} error={null} style={{ maxWidth: "100%" }}>
+          <EuiRadioGroup
+            options={radios}
+            idSelected={this.action.force_unsafe_input}
+            onChange={(id) => {
+              const forceUnsafe = id === "yes";
+              const shrinkObject = { ...action.action };
+              if (forceUnsafe) {
```
Comment (thalurur) [inline public/pages/VisualCreatePolicy/components/UIActions/ShrinkUIAction.tsx:216]: Not critical but may be doing a popup or having some warning when this is set to true stating can potentially result in data loss would be preferred I think

File: public/pages/VisualCreatePolicy/components/UIActions/ShrinkUIAction.tsx [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+
+  getIndexNameTemplateJsonString = (action: ShrinkAction) => {
+    if (action.hasOwnProperty("target_index_name_template_json")) {
+      return action.target_index_name_template_json;
+    } else {
+      return JSON.stringify(action.shrink.target_index_name_template, null, 4);
+    }
+  };
+
+  render = (action: UIAction<ShrinkAction>, onChangeAction: (action: UIAction<ShrinkAction>) => void) => {
```
Comment (thalurur) [inline public/pages/VisualCreatePolicy/components/UIActions/ShrinkUIAction.tsx:96]: Not sure, how we are doing for other actions but may be differentiating optional and required fields by an `*` or calling it out in words might be helpful

**Review by thalurur — APPROVED** [review]

File: public/pages/VisualCreatePolicy/components/UIActions/ShrinkUIAction.tsx (hunk 2 above)
Comment (downsrob) [reply public/pages/VisualCreatePolicy/components/UIActions/ShrinkUIAction.tsx:275]: I'll remove the console.log (yikes thanks for the catch) but deleting the jsons seems to be the pattern that we use for parsing jsons into action objects and deleting these jsons from the action copy so these extra parameters aren't sent to the backend

File: public/pages/VisualCreatePolicy/components/UIActions/ShrinkUIAction.tsx (hunk 4 above)
Comment (downsrob) [reply public/pages/VisualCreatePolicy/components/UIActions/ShrinkUIAction.tsx:96]: Added and demoed in a new screenshot

File: public/pages/VisualCreatePolicy/components/UIActions/ShrinkUIAction.tsx (hunk 3 above)
Comment (downsrob) [reply public/pages/VisualCreatePolicy/components/UIActions/ShrinkUIAction.tsx:223]: Added and demoed in a new screenshot

**Review by thalurur — APPROVED** [review]

**Review by bowenlan-amzn — APPROVED** [review]

**Pull request merged**